script/gen_requirements_all.py

```diff
@@ -266,7 +266,9 @@ def has_tests(module: str) -> bool:
     Test if exists: tests/components/hue/__init__.py
     """
     path = (
-        Path(module.replace(".", "/").replace("homeassistant", "tests")) / "__init__.py"
+        # Replace homeassistant/components to tests/components, but not homeassistant_hardware
+        Path(module.replace(".", "/").replace("homeassistant/", "tests/"))
```

Comment:
What about `.replace("homeassistant", "tests", 1)`? This should replace only the first instance, which seems to be the original intent of this code (since everything is namespaced under `homeassistant.` and needs to be rewritten to `tests.`).

Reply:
Done.